packages/sdk/src/types/FujiResult.ts

```diff
@@ -0,0 +1,5 @@
+import { FujiError } from '../entities/FujiError';
+
+export type FujiResult<T> =
+  | { success: true; data: T }
+  | { success: false; error: FujiError };
```

Comment:
What about creating 2 entities: `FujiResultError` and `FujiResultSuccess` (similar to `Token` and `NativeCurrency` that are both `Currency`):
```
export type FujiResult<T> = FujiResultSuccess<T> | FujiResultError;
```

That will allow to return `new FujiResultError(...)` and `new FujiResultSuccess(...)` instead of 

```
return {
  success: true,
  data: ....
}
```

Reply:
Initially, I thought it was a great idea but after quickly implementing it started having second thoughts.

First, it forces us to start typecasting or heavily extending. If we don't extend from the same class, we'll end up doing

```
if (result instanceof FujiResultSuccess) { 
  ... cast
}
```

Otherwise, we'll have to create yet another `FujiResultParent` class and have both `...Success` and `...Error` extend from it, then it would be a matter of checking for its properties. I think having a type with props would be more readable and way more standard.

Reply:
Wouldn't be enough just to check `if (result.success)`? And then inside being able to access `result.data` or `result.error` depending on the condition?

Reply:
You're right, I didn't realize it was a matter of adding the `success` property to each just like with the `type`.